From 701fd04d8a992bb9bbdba6ee18c7057c1d5c4674 Mon Sep 17 00:00:00 2001
From: Eugene Syromyatnikov <evgsyr@gmail.com>
Date: Sun, 19 Aug 2018 11:47:11 +0200
Subject: [PATCH 028/293] rtnl_link: add IFLA_BRPORT_BACKUP_PORT attribute

Introduced by Linux commit v4.19-rc1~140^2~288^2.

* xlat/rtnl_ifla_brport_attrs.in (IFLA_BRPORT_BACKUP_PORT): New
constant.
* rtnl_link.c (ifla_brport_nla_decoders) <[IFLA_BRPORT_BACKUP_PORT]>:
New decoder.
---
 rtnl_link.c                    | 3 ++-
 xlat/rtnl_ifla_brport_attrs.in | 1 +
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/rtnl_link.c b/rtnl_link.c
index 2421088..67ac7d1 100644
--- a/rtnl_link.c
+++ b/rtnl_link.c
@@ -171,7 +171,8 @@ static const nla_decoder_t ifla_brport_nla_decoders[] = {
 	[IFLA_BRPORT_BCAST_FLOOD]		= decode_nla_u8,
 	[IFLA_BRPORT_GROUP_FWD_MASK]		= decode_nla_u16,
 	[IFLA_BRPORT_NEIGH_SUPPRESS]		= decode_nla_u8,
-	[IFLA_BRPORT_ISOLATED]			= decode_nla_u8
+	[IFLA_BRPORT_ISOLATED]			= decode_nla_u8,
+	[IFLA_BRPORT_BACKUP_PORT]		= decode_nla_ifindex,
 };
 
 static bool
diff --git a/xlat/rtnl_ifla_brport_attrs.in b/xlat/rtnl_ifla_brport_attrs.in
index 74c8841..7017716 100644
--- a/xlat/rtnl_ifla_brport_attrs.in
+++ b/xlat/rtnl_ifla_brport_attrs.in
@@ -32,3 +32,4 @@ IFLA_BRPORT_BCAST_FLOOD		30
 IFLA_BRPORT_GROUP_FWD_MASK	31
 IFLA_BRPORT_NEIGH_SUPPRESS	32
 IFLA_BRPORT_ISOLATED		33
+IFLA_BRPORT_BACKUP_PORT		34
-- 
1.7.12.4

